src/components/core/header/app-header.tsx

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { File } from "@babel/types";
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { VscDebugRestart } from "react-icons/all";
+import { shallowEqual, useSelector } from "react-redux";
+import { getAuthenticators, updateAuthenticationSequence } from "../../../api";
+import Icon from "../../../assets/asgardeo-logo.svg";
+import { AuthenticationStep } from "../../../models";
+import { GenerateCodeFromAst } from "../../mapper";
+import { AlertModal } from "../modals";
+
+
+/**
+ * App header component.
+ *
+ * @return {React.ReactElement}
+ */
+export const AppHeader: FunctionComponent = () : ReactElement => {
+
+    const [authFactors, setAuthFactors] = useState<any>([]);
+    const [visibleSuccessAlertModal, setVisibleSuccessAlertModal] = useState<boolean>(false);
+    const [visibleErrorAlertModal, setVisibleErrorAlertModal] = useState<boolean>(false);
+    const searchUrlPrams = new URLSearchParams(window.location.search);
+    const appId: string|null = searchUrlPrams.get("appId");
+    const callbackUrl: string|null = searchUrlPrams.get("callbackUrl");
+
+    useEffect(() => {
+        getAuthenticators()
+            .then((response) => {
+                setAuthFactors(response.data);
+            })
+            .catch((error) => {
+                throw new Error(error);
```

Comment:
Let's add functionality to show these errors in the UI itself. We can track this as an improvement.

Reply:
Sure. Will track as an improvement.